### PR TITLE
Introduce a new Client Exception type to distinguish client\server separation when handling exception. 

### PR DESCRIPTION
**Pull request by NuwangaHerath**
Resolves wso2/product-is/issues/5369

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (somindatommy) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java:2]: Shall we fix the license year?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.exception;
+
+import org.wso2.carbon.identity.base.IdentityException;
+
+//ClientException class for distinguish client\server separation when handling exception in higher levels.
```
Comment (somindatommy) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java:23]: Shall we properly format the class comment?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java (hunk 1 above)
Comment (NuwangaHerath) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java]: Done

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java (hunk 2 above)
Comment (NuwangaHerath) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ClientException.java]: Done

**Pull request closed without merging**